---
title: "MicroMeta Vignette"
author: "Emily So 1004179131"
date: "November 17th 2020
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{MicroMetaVignette}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

```{r setup}
library(MicroMetaGraph)
```
## Introduction

`MicroMetaGraph` is an R package developed to demonstrate protein interaction networks given a set of input protein IDs. This includes the main components: DESCRIPTION, NAMESPACE, man subdirectory and R subdirectory. Additionally, licence, README and subdirectories vignettes, tests, data and inst are also explored. The package is targeted for metatranscriptomic researchs or those who are interested in protein interaction based on ortholog groups. **This document gives a tour of MicroMetaGraph (version 0.1.0)**. It was written in R Markdown, using the [knitr](https://cran.r-project.org/package=knitr) package for production. See `help(package = "MicroMetaGraph")` for further details. 

To download **MicroMetaGraph**, use the following commands:

``` r
require("devtools")
install_github("emilyso-99/MicroMetaGraph", build_vignettes = TRUE)
library("MicroMetaGraph")
```
To list all sample functions available in the package:
``` r
ls("package:MicroMetaGraph")
```

To list all sample datasets available in the package:
``` r
data(package = "MicroMetaGraph")
```

<br>

## Components

The components of this package is as follows:

<div style="text-align:center"><img src="SILVA_A_A1.png" alt="SILVA_A_A1" width="500"/>

<div style="text-align:left">

### Minimal requirements

* The **R subdirectory** contains all the R code files or script files for querying the relevant information and constructing the networks. These are files ending with ".R". 

* The **man subdirectory** contains help documentation for each function that is available in this package. These help files were autogenerated via *roxygen2* package based on the roxygen tags used in the .R files from R subdirectory.

* **DESCRIPTION** file contains metadata including the package name, the version number, the author and maintainer contact information, along with other information.

* **NAMESPACE** file provides a "context for looking up the value of an object associated with a name". This file will contain export(), import() and importFrom() statements. This file is auto-updated via *roxygen2* package.

### Other

* **README** file mainly describes what the package MicroMetaGraph does. For the purpose of this course, the following elements in order are required: 1. Description, 2. Installation, 3. Overview, 4. Contributions, 5. References, and 6. Acknowledgements.

* **LICENSE** file describing who can use the R package. 

* The **data** subdirectory with data (.rda) is included to demonstrate utility of functions in the package with this data. This was derived from the COG database, and the STRING metatranscriptomic databases

* The **tests** subdirectory includes the testthat subdirectory, which has test that are written to see if query and plot functions are working as expected.

* The **vignettes** subdirectory permit to include vignettes that provide an overview of the R package.

* The local project is housed in the owner's Github repository.The changelog representing the development of this package is presented through commit messages.

<br>

<div style="text-align:left">

## Package References

[So, E. (2020) MicroMetaGraph: A Package for Ortholog Protein Interaction Networks. Unpublished.](https://github.com/emilyso-99/MicroMetaGraph)

<br>

## Other References

R Core Team (2020). R: A language and environment for statistical computing. R Foundation for Statistical Computing, Vienna, Austria. https://www.R-project.org/

Wickham, H. and Bryan, J. (2019). *R Packages* (2nd edition). Newton, Massachusetts: O'Reilly Media. https://r-pkgs.org/

Tatusov RL, Galperin MY, Natale DA, Koonin EV. The COG database: a tool for genome-scale analysis of protein functions and evolution. Nucleic Acids Res. 2000;28(1):33-36. doi:10.1093/nar/28.1.33

Szklarczyk D, Gable AL, Lyon D, Junge A, Wyder S, Huerta-Cepas J, Simonovic M, Doncheva NT, Morris JH, Bork P, Jensen LJ, Mering CV. STRING v11: protein-protein association networks with increased coverage, supporting functional discovery in genome-wide experimental datasets. Nucleic Acids Res. 2019 Jan 8;47(D1):D607-D613. doi: 10.1093/nar/gky1131. PMID: 30476243; PMCID: PMC6323986.

----

```{r}
sessionInfo()
```